Source/Applications/SystemCenter/Controllers/ExternalDB/LineSegmentWizardController.cs

```diff
@@ -210,30 +210,31 @@ public IHttpActionResult PostData(int id, [FromBody] LineConfiguration record)
             Func<string> generatekey = () =>
             {
                 int i = 1;
-                string key = line.AssetKey + "-S" + i;
+                string key = line.AssetKey + "-S" + i.ToString("00");
                 TableOperations<Asset> assetTbl = new TableOperations<Asset>(connection);
                 while (assetTbl.QueryRecordCountWhere("Assetkey = {0}", key) > 0)
```

Comment:
If the table consists of unformatted AssetKeys, (i.e. -S1, -S2 instead of -S01, -S02), then this function might produce duplicate AssetKeys.

Reply:
Per our conversation:
(1) `Assetname` can not be overridden or the name the user sees won't match what we want
(2) `AssetName` in TSX file is empty
(3) `key` is used in the following few lines so it does need to be the actual `AssetKey`